### ISSUE TITLE
Confusing error if property setter function name is different than getter function name

### ISSUE THREAD

**Issue by scottbelden** [python/mypy#4234]
The following code produces a confusing error:

```python
class Foo:
    def __init__(self):
        self._foo = None

    @property
    def foo(self):
        return self._foo

    @foo.setter
    def bar(self, value):
        self._foo = value
```

Error:
```
foo.py:9: error: "Callable[[Any], Any]" has no attribute "setter"
```

I think this is related to https://github.com/python/mypy/issues/1465 but slightly different. Here my getter and setter are next to each other, but the function name is different. I'm not saying that having different function names is good practice (I meant for them to be the same, but I had a typo), but technically it is valid code and shouldn't produce an error:

```python
>>> import foo
>>> f = foo.Foo()
>>> f.foo
>>> f.bar = 1
>>> f.foo
1
```

I like the fact that mypy accidentally caught this typo, but I don't think it should have.

**Comment by gvanrossum** [comment]
What's technically valid and what mypy should allow are two different things. So if this is bad style I like that mypy caught it. If you really want your code like that all you have to do is add `# type: ignore` to the line with the error.

**Comment by scottbelden** [comment]
I agree. I guess I would just hope that it might be possible for the error message to be a little more clear that the problem is in the function name.
